src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -22,11 +22,16 @@ public sealed partial class ModuleMetadata : Metadata
         private bool _isDisposed;
 
         private readonly PEModule _module;
+        
+        private readonly UnmanagedMemoryStream? _memoryStream;
```

Comment:
>UnmanagedMemoryStream [](http://example.com/codeflow?start=25&length=21)

You could keep the previous concept of IDisposable owner and `_disposeOwner` flag here since we only need this field for keeping the object alive and disposing it.